model_analyzer/result/model_config_measurement.py

```diff
@@ -0,0 +1,283 @@
+# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from model_analyzer.constants import COMPARISON_SCORE_THRESHOLD
+from model_analyzer.constants import LOGGER_NAME
+
+from model_analyzer.record.record import RecordType
+
+from functools import total_ordering
+import logging
+
+logger = logging.getLogger(LOGGER_NAME)
+
+
+@total_ordering
+class ModelConfigMeasurement:
+    """
+    Encapsulates the set of metrics obtained from a single model config's
+    RunConfig run
+    """
+
+    def __init__(self, model_config_name, model_specific_pa_params,
+                 non_gpu_data):
+        """
+        model_config_name : string
+            The model config name that was used in the RunConfig
+        model_specific_pa_params: dict
+            Dictionary of PA parameters that can change between models
+            in a multi-model RunConfig
+        non_gpu_data : list of Records
+            Metrics that do not have a GPU UUID associated with them,
+            from either CPU or PA
+        """
+
+        self._model_config_name = model_config_name
+        self._model_specific_pa_params = model_specific_pa_params
+        self._non_gpu_data = non_gpu_data
+
+        self._non_gpu_data_from_tag = {
+            type(metric).tag: metric for metric in self._non_gpu_data
+        }
+
+    @classmethod
+    def from_dict(cls, model_config_measurement_dict):
+        model_config_measurement = ModelConfigMeasurement(None, {}, [])
```

Comment:
Does it make sense to have a second constructor with no args that calls the constructor with these args? So the user doesn't have to know to pass these special empty variables?